### PR TITLE
 General extension to open an external link

### PR DESCRIPTION
**Pull request by dqdquint**
The ticket originally wanted to only add a predefined image and link to open an external page, but I went ahead and decided to make a general extension for this feature.

### CODE REVIEW

File: extension-config.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+  "version": "1.0.0",
+  "id": "@shopgate-project/open-external-link",
+  "components": [
+    {
+      "id": "externalLink",
```
Comment (Carsten89) [inline extension-config.json]:
```suggestion
      "id": "ExternalLink",
```
afaik other widget use upper case first letter

File: frontend/widgets/ExternalLink/connector.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { connect } from 'react-redux';
+import { openPageExtern } from '@shopgate/engage/core';
+
+/**
+ * @returns {Object}
+ */
+const mapStateToProps = () => ({
+  openPageExtern: (params) => {
+    openPageExtern(params);
```
Comment (Carsten89) [inline frontend/widgets/ExternalLink/connector.js]: openPageExtern is a simple function. so there is no need for mapStateToProps here. 
mapStateToProps should only be used to map the redux state to component props, which is not the case here

File: frontend/widgets/ExternalLink/index.jsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import connect from './connector';
+import styles from './style';
+
+/**
+ * @returns {JSX}
+ */
+const ExternalLink = ({ settings, openPageExtern }) => {
+  const {
+    imgSource, linkText, link, imageHeight, imageWidth, fontSize,
+  } = settings || null;
```
Comment (Carsten89) [inline frontend/widgets/ExternalLink/index.jsx]: if your `|| null` would really happen here, the code would through a JS error. "can not read prop imgSource of null".
if you want to add a check ---> `|| {}` is the correct way

File: frontend/widgets/ExternalLink/index.jsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  }
+
+  /**
+   * Opens external page
+   */
+  const handleClick = () => {
+    openPageExtern({ src: link });
+  };
+
+  const imageLink = (
```
Comment (Carsten89) [inline frontend/widgets/ExternalLink/index.jsx]: add check if there is a imgSource set before create this here

File: frontend/widgets/ExternalLink/index.jsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    <button onClick={handleClick}>
+      <img className={styles.image(imageHeight, imageWidth)} alt="Configurabale" src={imgSource} />
```
Comment (Carsten89) [inline frontend/widgets/ExternalLink/index.jsx]: typo Configurable.
and not sure if this alt text will really add any value here

File: README.md [hunk 6]
```diff
@@ -1,13 +1,35 @@
...
+
+## About Widget Configuration
+The widget can display either a link in text or a image link. There are additional optional configuration for the styling of either the image or text link.
+
+- `link`: the url to be opened in an external browser
+####Use one of the following two configurations.
+- `imgSource`: the source url for your image link
+- `linkText`: the text to display for your text link
+####The following configurations are optional
+- `imageHeight`: the specified height for your image
```
Comment (Carsten89) [inline README.md]: we should mention that here all possible css values for height and width are allowed. So %, rem, 'auto' etc

**Review by Carsten89 — CHANGES_REQUESTED** [review]

File: extension-config.json (hunk 1 above)
Comment (dqdquint) [reply extension-config.json]: Updated

File: frontend/widgets/ExternalLink/connector.js (hunk 2 above)
Comment (dqdquint) [reply frontend/widgets/ExternalLink/connector.js]: Gotcha. Updated and tested.

File: frontend/widgets/ExternalLink/index.jsx (hunk 3 above)
Comment (dqdquint) [reply frontend/widgets/ExternalLink/index.jsx]: Updated. Thanks

File: frontend/widgets/ExternalLink/index.jsx (hunk 5 above)
Comment (dqdquint) [reply frontend/widgets/ExternalLink/index.jsx]: I agree that it doesn't have any value, but eslint threw an error without an alt attribute, and disabling for the entire file didn't seem worthwhile.

File: frontend/widgets/ExternalLink/index.jsx (hunk 4 above)
Comment (dqdquint) [reply frontend/widgets/ExternalLink/index.jsx]: Updated

File: README.md (hunk 6 above)
Comment (dqdquint) [reply README.md]: Updated

**Review by Carsten89 — APPROVED** [review]

**Pull request merged**